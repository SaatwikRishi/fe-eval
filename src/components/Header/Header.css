header {
    height: 100px;
    font-size: 40px;
    color: white;
    background-color: black;
    display: flex;
    font-family: "Big Shoulders Display" !important;
}
header > p {
    align-self: center;
    padding-left: 20px;
    
}
