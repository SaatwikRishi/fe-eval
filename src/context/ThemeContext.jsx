import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { makeRequest } from '../util/makeRequest'
import { GET_THEMES } from '../util/constants'
import { useNavigate } from 'react-router-dom'

export const ThemeContext = React.createContext()

export default function ThemeState ({ children }) {
  const [themes, setThemes] = React.useState([])
  const [loading, setLoading] = React.useState(true)
  const [currentTheme, setCurrentTheme] = React.useState(1)
  const getCurrentTheme = () => {
    const th = themes.filter((e) => e.id === currentTheme)[0].colorHexCode
    return th
  }

  const navigate = useNavigate()

  useEffect(() => {
    makeRequest(GET_THEMES())
      .then((e) => {
        setCurrentTheme(e.preferredThemeId)

        setThemes(e.themes)
        setTimeout(() => {
          setLoading(false)
        })
      }, 300)
      .catch((e) => {
        navigate('/error/500')
        setLoading(false)
      })
  }, [])
  if (loading) return <div>Loading</div>
  return (
    <ThemeContext.Provider value={{ themes, setCurrentTheme, getCurrentTheme }}>
      {children}
    </ThemeContext.Provider>
  )
}

ThemeState.propTypes = {
  children: PropTypes.node.isRequired
}
