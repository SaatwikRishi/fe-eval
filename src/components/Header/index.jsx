import React, { useContext } from 'react'
import { ThemeContext } from '../../context/ThemeContext'
import './Header.css'

export default function Header () {
  // eslint-disable-next-line no-unused-vars
  const { getCurrentTheme } = useContext(ThemeContext)
  console.log(getCurrentTheme())
  return (
    <header className="paddings" style={{
      backgroundColor: getCurrentTheme()
    }}>

      <p>EVENTIFY</p>
    </header>
  )
}
