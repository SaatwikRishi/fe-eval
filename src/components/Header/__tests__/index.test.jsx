import { render } from '@testing-library/react'
import Header from '..'
import React from 'react'
import { ThemeContext } from '../../../context/ThemeContext'

describe('Tests for Header', () => {
  it('should render when component is called', () => {
    const { container } = render(
      <ThemeContext.Provider value={{ getCurrentTheme: jest.fn() }}>

    <Header />
    </ThemeContext.Provider>

    )
    expect(container).toMatchSnapshot()
  })
})
