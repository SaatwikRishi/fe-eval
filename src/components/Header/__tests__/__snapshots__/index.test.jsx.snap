// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Header should render when component is called 1`] = `
<div>
  <header
    class="paddings"
  >
    <p>
      EVENTIFY
    </p>
  </header>
</div>
`;
